### PR TITLE
Web3 integration

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +46,7 @@
     "react-select": "^2.1.0",
     "react-timezone": "^2.3.0",
     "react-tooltip": "^3.9.0",
+    "reactive-properties": "^0.1.5",
```
Comment (marekkirejczyk) [inline package.json]: Do we need `reactive-properties`?

File: src/api/web3.js [hunk 2]
```diff
@@ -147,6 +148,42 @@ const getWeb3 = lazyAsync(async () => {
...
+    new Web3Strategy(
+      [
+        {
+          name: 'UniversalLogin',
+          icon: 'UniversalLogin logo',
+          create: async () => {
+            const ulProvider = new ULWeb3Provider({
+              provider: new Web3.providers.HttpProvider(
+                'https://kovan.infura.io/v3/b3026fc5137a4bd18e5d5906ed49f77d'
+              ),
```
Comment (xitronix) [inline src/api/web3.js]: use provider from line 152 instead

File: src/api/web3.js [hunk 3]
```diff
@@ -126,7 +127,7 @@ const getWeb3 = lazyAsync(async () => {
       networkState.allGood = false
       throw new Error('Error setting up web3')
     }
-
+    setupWeb3(web3)
```
Comment (xitronix) [inline src/api/web3.js]: Unilogin.setupWeb3(web3, 'UniLogin', {...});
Unilogin.setupWeb3(web3, ['UniLogin', 'Metamask'], {...});
Unilogin.setupWeb3(web3, ['UniLogin', 'Metamask', {}], {...});

File: src/api/web3.js [hunk 4]
```diff
@@ -147,6 +148,42 @@ const getWeb3 = lazyAsync(async () => {
...
+              }
+            })
+            await ulProvider.init()
+            return ulProvider
+          }
+        },
+        {
+          create: () => provider,
+          icon: 'Kickback icon',
+          name: 'Kickback default provider'
```
Comment (xitronix) [inline src/api/web3.js]: Default provider

File: src/api/web3.js (hunk 3 above)
Comment (xitronix) [reply src/api/web3.js]: const options = {
  applicationInfo: {
    applicationName: 'Kickback',
    logo: 'https://kickback.events/favicon.ico',
    type: 'laptop'
  }
}

File: src/api/web3.js [hunk 5]
```diff
@@ -147,6 +148,42 @@ const getWeb3 = lazyAsync(async () => {
...
+      [
+        {
+          name: 'UniversalLogin',
+          icon: 'UniversalLogin logo',
+          create: async () => {
+            const ulProvider = new ULWeb3Provider({
+              provider: new Web3.providers.HttpProvider(
+                'https://kovan.infura.io/v3/b3026fc5137a4bd18e5d5906ed49f77d'
+              ),
+              relayerUrl: 'https://relayer-kovan.herokuapp.com',
```
Comment (xitronix) [inline src/api/web3.js]: use web3 to discover network and propose proper relayer url

File: src/api/web3.js [hunk 6]
```diff
@@ -147,6 +148,42 @@ const getWeb3 = lazyAsync(async () => {
...
+              ensDomains: ['poppularapp.test'],
```
Comment (xitronix) [inline src/api/web3.js]: ensDomains - get from server

File: src/components/Header/SignInButton.js [hunk 7]
```diff
@@ -33,6 +34,11 @@ function SignInButton() {
     networkState,
     reloadUserAddress
   }) => async () => {
+    const web3 = await getWeb3()
+    const { waitForPick } = web3.currentProvider.web3picker.show()
+    await waitForPick()
+    networkState.readOnly = false
```
Comment (xitronix) [inline src/components/Header/SignInButton.js]: 1. `await UniLogin.getWeb3Picker(web3).showAndWait();`

**Pull request closed without merging**